game/src/main/java/net/demilich/metastone/game/logic/GameLogic.java

```diff
@@ -3493,7 +3526,20 @@ public void startTurn(int playerId) {
 			refreshAttacksPerRound(minion);
 		}
 		context.fireGameEvent(new TurnStartEvent(context, player.getId()));
-		drawCard(playerId, null);
+
+		boolean shouldDraw = true;
+		for (Trigger e : context.getTriggersAssociatedWith(player.getReference())) {
+			if (e instanceof Enchantment) {
+				Enchantment enchantment2 = (Enchantment) e;
```

Comment:
You can replace `drawCard` here with a call to `DrawCardSpell` so you don't have to special case `enchantment_frozen_deck`. Eventually, we're going to replace this behaviour with a "rule card" that can be silenced.